kickoff/static/suggestions.js

```diff
@@ -121,16 +125,46 @@ function populatePartial(name, version, previousBuilds, partialElement) {
         } else {
             previousReleases = previousBuilds[base + 'release'].sort().reverse()
         }
+
+        if (isESR(version)) {
+            // Use the ESR partial
+            partialsADI = partialsADIESR;
+        } else {
+            partialsADI = partialsADIRelease;
+        }
+        // For thunderbird, use only the three last
+
         nbPartial = 3;
     }
 
+    // Transform the partialsADI datastruct in a single array to
+    // simplify processing
+    partialsADIVersion = [];
+    for (i = 0; i < partialsADI.length; i++) {
+        partialsADIVersion[i] = partialsADI[i].version;
+    }
+
     partial = "";
     partialAdded = 0;
     for (i = 0; i < previousReleases.length; i++) {
+
         // Reuse the previous builds info to generate the partial
         // Strip the build number. "31.0build1" < "31.0.1" => false is JS
         previousRelease = previousReleases[i].replace(/build.*/g, "");
-        if (previousRelease < version) {
+
+        // Some explanations on this code
+        // We want:
+        // * the last version published (no matter the ADI).
+        //   This because the ADI takes some time to be relevant
+        // * we take the versions which match the highest ADIs
+        // * For TB, uses only the X last builds
+        // * As we don't have the detail partial for beta. For the use of the X last builds
+        hasImportantADI = partialsADIVersion.indexOf(previousRelease.replace("esr","")) > -1;
+        if (previousRelease < version && ( hasImportantADI ||
+                                           i == 0 ||
+                                           isTB(name) ||
+                                           isBeta(version)
+                                         )) {
```

Comment:
I'm not sure this code is going to work with larger sets of data. AFAICT, we'll end up with partials for ALL old Firefox and Thunderbird versions, because:
* hasImportantADI will be true for every Firefox release
* isTB(name) will be true for every Thunderbird release
* isBeta(version) will be true for all Betas.

Please let me know if I'm missing something here. It seems like this code could be refactored to be much cleaner by doing something like:
* Add the most recent release outside of it
* Sort the partial ADI data structure by ADI
* Add the first N-1 items in it to the list of partials (where N could be different per product or release type, and we subtract 1 to account for the previous version that was already added).

There's definitely a bug in the existing code where you'll end up adding the wrong versions. Eg, with ADI like:
* 36.0.4 - 5000
* 36.0.3 - 5000
* 36.0 - 500
* 35.0 - 3000

You'd add 36.0 instead of 35.0, because they both have "important" ADI, even though 35.0 has more than 36.0.